### PR TITLE
Bootstrap: disable logging for missing XCTRequest

### PR DESCRIPTION
**Pull request by jmoody**
**REBASED** Thu Sep 1 19:29 CET
### Motivation

With the Springboard alert handler in the RunLoop the log fills up instantly with these message.  The handler logs and returns nil so I think it is fine to turn off the logging.

Pages and pages of this:

```
2016-09-01 00:07:12.638 iOSDeviceManager[55759:536058] *** Assertion failure in -[FBTestManagerAPIMediator handleUnimplementedXCTRequest:], /Users/moody/git/calabash/FBSimulatorControl/XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m:386
2016-09-01 00:07:12.639 iOSDeviceManager[55759:536058] _XCT_testCase:method:didFinishActivity:
2016-09-01 00:07:12.639 iOSDeviceManager[55759:536058] Received call for unhandled method (_XCT_testCase:method:didFinishActivity:). Probably you should have a look at _IDETestManagerAPIMediator in IDEFoundation.framework and implement it. Good luck!
2016-09-01 00:07:12.639 iOSDeviceManager[55759:536058] *** Assertion failure in -[FBTestManagerAPIMediator handleUnimplementedXCTRequest:], /Users/moody/git/calabash/FBSimulatorControl/XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m:386
2016-09-01 00:07:12.736 iOSDeviceManager[55759:536058] _XCT_testCase:method:willStartActivity:
2016-09-01 00:07:12.736 iOSDeviceManager[55759:536058] Received call for unhandled method (_XCT_testCase:method:willStartActivity:). Probably you should have a look at _IDETestManagerAPIMediator in IDEFoundation.framework and implement it. Good luck!
2016-09-01 00:07:12.736 iOSDeviceManager[55759:536058] *** Assertion failure in -[FBTestManagerAPIMediator handleUnimplementedXCTRequest:], /Users/moody/git/calabash/FBSimulatorControl/XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m:386
2016-09-01 00:07:12.848 iOSDeviceManager[55759:536058] _XCT_testCase:method:willStartActivity:
2016-09-01 00:07:12.848 iOSDeviceManager[55759:536058] Received call for unhandled method (_XCT_testCase:method:willStartActivity:). Probably you should have a look at _IDETestManagerAPIMediator in IDEFoundation.framework and implement it. Good luck!
```

### CODE REVIEW

**Comment by jmoody** [comment]
Was not enough.  Now I see:

```
2016-09-01 01:29:07.598 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:07.598 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:07.772 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:07.799 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:07.847 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:07.847 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:07.847 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:07.847 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.021 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:08.048 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:08.101 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.101 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:08.101 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.101 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.274 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:08.303 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:08.352 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.353 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
2016-09-01 01:29:08.353 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.353 iOSDeviceManager[67425:591881] _XCT_testCase:method:didFinishActivity:
2016-09-01 01:29:08.523 iOSDeviceManager[67425:591881] _XCT_testCase:method:willStartActivity:
```

**Comment by jmoody** [comment]
Much better.

We still see logs from the start and terminate test plan (ones that we might care about), but not others.

**Comment by sapieneptus** [comment]
@jmoody Somewhere deep in the heart of the beast is an Env Var that dictates whether to use verbose logging or not. Are you sure we don't want to wrap these log statements in an `if (VERBOSE_VAR) {}` block instead of killing them completely?

**Comment by jmoody** [comment]
**REBASED** Thu Sep 1 19:29 CET

Ready for review.

**Comment by jmoody** [comment]
@sapieneptus I want to merge.  I have another change in the wings (adding some useful logging about test failures).

**Comment by sapieneptus** [comment]
K, but we never settled the point I raised.

**Comment by jmoody** [comment]
It is settled?  I did what you suggested?

File: XCTestBootstrap/TestManager/FBXCTestManagerLoggingForwarder.m [hunk 1]
```diff
@@ -56,7 +56,11 @@ - (NSMethodSignature *)methodSignatureForSelector:(SEL)selector
 - (void)forwardInvocation:(NSInvocation *)invocation
 {
   if ([self.interface respondsToSelector:invocation.selector]) {
-    [self.logger.debug log:NSStringFromSelector(invocation.selector)];
+
+    if ([FBControlCoreGlobalConfiguration debugLoggingEnabled]) {
```
Comment (jmoody) [inline XCTestBootstrap/TestManager/FBXCTestManagerLoggingForwarder.m:60]: This is controlled by one of the variables that iOSDeviceManager `VERBOSE` sets.

**Comment by sapieneptus** [comment]
You did? I don't see any changes in the code. I was asking if you thought we should wrap these in an `if (VERBOSE) {  }` or something

File: XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m [hunk 2]
```diff
@@ -301,12 +301,12 @@ - (id)_XCT_recordedEventNames:(NSArray *)arg1 timestamp:(NSNumber *)arg2 duratio
 
 - (id)_XCT_testCase:(NSString *)arg1 method:(NSString *)arg2 didFinishActivity:(XCActivityRecord *)arg3
 {
-  return [self handleUnimplementedXCTRequest:_cmd];
+  return nil;
```
Comment (sapieneptus) [inline XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m:304]: @jmoody I guess more specifically I meant this as well. Should it 

``` objc
if (Verbose) {  return [self handleUnimplementedXCTRequest:_cmd]; } 
return nil
```

?

Comment (jmoody) [reply XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m:304]: I don't think so, but if you want that behavior I will add it.

Comment (sapieneptus) [reply XCTestBootstrap/TestManager/FBTestManagerAPIMediator.m:304]: I was asking your opinion. If you feel like there's _never_ a time we'd want this log, then it's fine.

**Pull request merged**